System/Win32/Console.hsc

```diff
@@ -154,3 +163,65 @@ getCurrentConsoleScreenBufferInfoEx :: IO CONSOLE_SCREEN_BUFFER_INFOEX
 getCurrentConsoleScreenBufferInfoEx = do
     h <- failIf (== nullHANDLE) "getStdHandle" $ getStdHandle sTD_OUTPUT_HANDLE
     getConsoleScreenBufferInfoEx h
+
+
+-- c_GetEnvironmentVariableW :: LPCWSTR -> LPWSTR -> DWORD -> IO DWORD
+getEnv :: String -> IO (Maybe String)
+getEnv name =
+  withCWString name $ \c_name -> withTStringBufferLen maxLength $ \(buf, len) -> do
+    let c_len = fromIntegral len
+    c_len' <- c_GetEnvironmentVariableW c_name buf c_len
+    if c_len' == 0
+    then do
+      err_code <- getLastError
+      if err_code  == eERROR_ENVVAR_NOT_FOUND
+      then return Nothing
+      else errorWin "GetEnvironmentVariableW"
+    else do
+      let len' = fromIntegral c_len'
+      Just <$> peekTStringLen (buf, len')
+ where
+  -- according to https://learn.microsoft.com/en-us/windows/win32/api/processenv/nf-processenv-getenvironmentvariablew
+  maxLength :: Int
+  maxLength = 32767

```

Comment:
Documentation says "characters". Do  they mean wide chars? Then we have to *2 this, I guess.

Reply:
Yeah, the buffer size is 16-bits, so 2^16-1 bytes.